pandas/_libs/testing.pyx

```diff
@@ -180,13 +180,15 @@ cpdef assert_almost_equal(a, b,
         # classes can't be the same, to raise error
         assert_class_equal(a, b, obj=obj)
 
-    if a == b:
-        # object comparison
-        return True
     if isna(a) and isna(b):
         # TODO: Should require same-dtype NA?
```

Comment:
this might be a reasonable time to start enforcing same-NA

Reply:
Open a separate issue for it?